scripts/check-dev-utils.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+cd "$(dirname "$0")/.."
+source ci/_
+# only nightly is used uniformly as we contain good amount of nightly-only code
+# (benches, frozen abi...)
+source ci/rust-version.sh nightly
+
+# There's a special common feature called `dev-utils` to
```

Comment:
actually maybe a `use-` or `export-` would better self-describe as well :thinking:

Reply:
> can we call the feature `test-utils` instead? or something that implies that the feature gates test-specific logic.

well i avoided `test` because of existence of `benches`. so, i followed cargo's `dev-dependency` (it isn't called `test-dependency`).

>  maybe `sol{,ana}`-prefixed

hmm, i don't feel strong for prefixing as other existing features aren't prefixed. also, features are usually namespaced to workspace. so, even if we use `--features XXX`, it won't enable features from other crates. say, `--features serde` on CLI doen't enable `im`'s `serde` feature. so, I wonder what's justification of prefixing just for `dev-utils` in that regard too.

> actually maybe a `use-` or `export-` would better self-describe as well :thinking:

i think features themselves imply _using_ / _exporting_ as cargo's features are additive by design. this new feature (`dev-utils`) works in additive manner just like all other normal features. we're just special-casing it to prevent accidental overuse and misuse at CI layer.

Reply:
oh _that_ `dev-` :sweat: 

i think we need another word in the middle to disambiguate. will think on it some more